crates/rome_js_analyze/src/lib.rs

```diff
@@ -341,3 +357,81 @@ impl std::fmt::Display for RuleError {
 }
 
 impl Error for RuleError {}
+
+/// We now try to "guess" the token where to apply the suppression comment.
+/// Considering that the detection of suppression comments in the linter is "line based", we start
+/// querying the node covered by the text range of the diagnostic, until we find the first token that has a newline
+/// among its leading trivia.
+///
+/// If we're not able to find any token, it means that the range is
+/// placed at row 1, so we take the root itself.
+fn apply_suppression_comment(payload: SuppressionCommentEmitterPayload<JsLanguage>) {
+    let SuppressionCommentEmitterPayload {
+        token_offset,
+        mutation,
+        suppression_text,
+        diagnostic_text_range,
+    } = payload;
+    let current_token = match token_offset {
+        TokenAtOffset::None => None,
+        TokenAtOffset::Single(token) => Some(match find_token_with_newline(token.clone()) {
+            None => token,
+            Some(token) => token,
+        }),
+        TokenAtOffset::Between(left_token, right_token) => {
+            let chosen_token = if right_token.text_range().start() == diagnostic_text_range.start()
+            {
+                right_token
+            } else {
+                left_token
+            };
+            Some(chosen_token)
+        }
+    };
+    if let Some(current_token) = current_token {
+        if let Some(element) = current_token.ancestors().find_map(JsxAnyChild::cast) {
+            let jsx_comment = jsx_expression_child(
+                token(T!['{']).with_trailing_trivia([(
+                    TriviaPieceKind::SingleLineComment,
+                    format!("/* {} */", suppression_text).as_str(),
+                )]),
+                token(T!['}']),
+            );
+            mutation.add_jsx_element_before_element(
+                &element,
+                &JsxAnyChild::JsxExpressionChild(jsx_comment.build()),
+            );
+        } else {
+            let new_token = current_token.with_leading_trivia(vec![
```

Comment:
You can use an array instead of a vec
```suggestion
            let new_token = current_token.with_leading_trivia([
```